### DISCUSSION TITLE
[FEATURE] support node archived package

### DISCUSSION THREAD

**Discussion by skyshore2001** [npm/rfcs#100] [General]
# What / Why

The node_modules folder contains too many files, which causes huge disk space and disk operation cost when install packages, load modules and even delete the folder.

I create an template Vue project,  and get 40000 files, 350MB disk space for the node_modules folder.
I  attempt to zip the folder and get a 78MB zip file. (compress rate ~20%).

## How

npm/node is supposed to suppthe the archived package like jar for java package.
e.g. the core-js package contains >1700 files and 7MB, it can be installed as one file like `core-js.3.0.1.nar` with <500KB space.
node is supposed to load the archived package in-memory instead of loading hundreds of js files, and I believe it also benefit the performance greatly.

**Comment by hihurmuz** [comment]
So you are talking about a project structure that uses npm packages in common? This is a great idea. is there a way for that?

**Comment by hihurmuz** [comment]
Packages should only be uploaded to projects in the production environment and used common in the development environment.

**Comment by ljharb** [comment]
node still needs to load each file separately.
